### ISSUE TITLE
Elasticsearch AWSRequestMetrics is missing ServiceName

### ISSUE THREAD

**Issue by shawnsmith** [aws/aws-sdk-java#1052]
Since the recent json protocol refactoring in 1.11.94, some `AWSRequestMetrics` objects are missing a value for `AWSRequestMetrics.Field.ServiceName`.  This breaks my existing custom metrics reporting code.

For example, the following code fragment used to print `Service name: [AWSElasticsearch]` and now it prints `Service name: [null]`:
```
AWSElasticsearch es = AWSElasticsearchClientBuilder.standard()
        .withRegion("us-east-1")
        .withMetricsCollector(new RequestMetricCollector() {
            @Override
            public void collectMetrics(Request<?> request, Response<?> response) {
                AWSRequestMetrics metrics = request.getAWSRequestMetrics();
                List<Object> serviceName = metrics.getProperty(AWSRequestMetrics.Field.ServiceName);
                System.out.println("Service name: " + serviceName);
            }
        })
        .build();
es.describeElasticsearchDomain(new DescribeElasticsearchDomainRequest().withDomainName("foo"));
```

At first glance, it looks like the problem might be in `RequestMarshaller.ftl` which has the following:
```
private static final OperationInfo SDK_OPERATION_BINDING = OperationInfo.builder()
    ...
    <#if shape.marshaller.target??>
    .operationIdentifier("${shape.marshaller.target}")
    .serviceName("${serviceNameForRequest}")
    </#if>
    .build();
```
Is the scope of the `<#if>` correct?  Or should it just protect the reference to `${shape.marshaller.target}` and not the line with `${serviceNameForRequest}`.

**Comment by kiiadi** [comment]
Good catch! We'll have a fix out for this in the next couple of days.

**Comment by kiiadi** [comment]
@shawnsmith fix for this went out last night in 1.11.102
